src/Core/Helpers.cs

```diff
@@ -61,11 +61,12 @@ private static HttpRequestMessage CreateProxyHttpRequest(this HttpContext contex
             var requestMethod = request.Method;
 
             // Copy the request headers.
-            if (requestMessage.Content != null)
+            foreach (var header in request.Headers)
             {
-                foreach (var header in request.Headers)
-                    if (!requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray()))
-                        requestMessage.Content?.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray());
+                if (!requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray()))
+                {

```

Comment:
Please remove extraneous braces.  Also, please just edit this branch so we don't lose the comments.